examples/llm_plus_gnn/rag_graph_store.py

```diff
@@ -0,0 +1,92 @@
+from typing import Optional, Union
+
+import torch
+from torch import Tensor
+
+from torch_geometric.data import FeatureStore
+from torch_geometric.distributed import LocalGraphStore
+from torch_geometric.sampler import (
+    HeteroSamplerOutput,
+    NeighborSampler,
+    NodeSamplerInput,
+    SamplerOutput,
+)
+from torch_geometric.sampler.neighbor_sampler import NumNeighborsType
+from torch_geometric.typing import EdgeTensorType, InputEdges, InputNodes
+
+
+class NeighborSamplingRAGGraphStore(LocalGraphStore):
+    def __init__(self, feature_store: Optional[FeatureStore] = None,
+                 num_neighbors: NumNeighborsType = [1], **kwargs):
+        self.feature_store = feature_store
+        self._num_neighbors = num_neighbors
+        self.sample_kwargs = kwargs
+        self._sampler_is_initialized = False
+        super().__init__()
+
+    def _init_sampler(self):
+        if self.feature_store is None:
+            raise AttributeError("Feature store not registered yet.")
+        self.sampler = NeighborSampler(data=(self.feature_store, self),
+                                       num_neighbors=self._num_neighbors,
+                                       **self.sample_kwargs)
+        self._sampler_is_initialized = True
+
+    def register_feature_store(self, feature_store: FeatureStore):
+        self.feature_store = feature_store
+        self._sampler_is_initialized = False
+
+    def put_edge_id(self, edge_id: Tensor, *args, **kwargs) -> bool:
+        ret = super().put_edge_id(edge_id.contiguous(), *args, **kwargs)
+        self._sampler_is_initialized = False
+        return ret
+
+    @property
+    def edge_index(self):
```

Comment:
this is a feature, not example, integrate into torch_geometric package itself

Reply:
The FeatureStore and Graphstore defined here are meant to be PoCs and cannot scale. I want this example to be a starting point, so that if someone writes to write a FeatureStore/GraphStore interface for a 3rd party service such as Neo4j/Pinecone/etc., they have a starting point. The constraints for the RAG interface are defined by the protocols found in rag_loader.py